modules/motions/ColorStraight.cpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * @file   ColorStraight.cpp
+ * @brief  目標色まで直進するクラス
+ * @author HaruArima08
+ */
+
+#include "ColorStraight.h"
+
+ColorStraight::ColorStraight(Robot& _robot, COLOR _targetColor, double _targetSpeed)
+  : Straight(_robot, _targetSpeed), targetColor(_targetColor)
+{
+}
+
+bool ColorStraight::isMetPreCondition()
+{
+  // 目標速度が0の場合は終了する
+  if(targetSpeed == 0) {
```

Comment:
```suggestion
  if(targetSpeed == 0.0) {
```
targetSpeed は double なので 0.0 で。